### PR TITLE
[auth][needs-docs][feature] Clear access cache

### PR DESCRIPTION
**Pull request by elpaso**
## Description
This fixes a bug that required to restart
QGIS after a connection failed due to SSL
errors (the connection auth was cached).

## Snapshots

options-cache-2
options-cache-1
options-cache-3

## Also

- added a tab for content and connection
  auth cache configuration
- user feedback on (both) caches cleared
- changed icon for content cache clear to
  a trash bin (after users enquiries about
  what was that "back/undo" icon for)

### CODE REVIEW

File: src/gui/auth/qgsautheditorwidgets.cpp [hunk 1]
```diff
@@ -147,6 +148,11 @@ void QgsAuthEditorWidgets::setupUtilitiesMenu()
   mActionRemoveAuthConfigs = new QAction( QStringLiteral( "Remove all authentication configurations" ), this );
   mActionEraseAuthDatabase = new QAction( QStringLiteral( "Erase authentication database" ), this );
 
+  mActionClearAccessCacheNow = new QAction( QStringLiteral( "Clear network authentication access cache" ), this );
```
Comment (m-kuhn) [inline src/gui/auth/qgsautheditorwidgets.cpp]: Translate?

**Comment by m-kuhn** [comment]
Out of curiosity, when will I want to disable this setting?

**Comment by dakcarto** [comment]
@m-kuhn wrote:
> Out of curiosity, when will I want to disable this setting?

Unfortunately, Qt does not offer a means of clearing _only_ the cached authentication for a particular endpoint you are trying to access, i.e. for host:port, etc. Which means ALL items in a network connection authentication cache (per `QgsNetworkAccessManager` instance here, I believe) are *removed*. This may have undesired effects (further testing needed with large projects and multithreading here to verify that).

We _know_ it helps with authenticated SSL connections, where currently invalid connection authentication is cached and always reused on future attempts, even after _correct_ credentials are substituted. However, when @elpaso and I discussed whether this should be ON by default, it made sense, as long as there was a way to turn it off, if needed.

Basically, IMO such a setting, which potentially affects many similarly authenticated layers in a big project, needs a means of turning it off until it can be determined, through extensive real-world usage, that it can be left always ON programmatically. 

To answer you question... I can't think of a specific case for turning it OFF, but I'm sure someone will find a use case or two for us 😉

**Comment by m-kuhn** [comment]
I would certainly have liked to remove this option (we've got too many) but that sounds convincing.
My main concern is, that it's very hard to tell for anyone who didn't implement this feature why you would (not) want this enabled and what the default state would be. Given that we don't know in which scenario this should be turned off, it's hard to give clues about this (afaict) but possibly appending a "(recommended)" and a tooltip explaining in which situation it helps will allow users to make an informed decision?

**Comment by elpaso** [comment]
@m-kuhn 
clear-cache-tooltip

**Comment by m-kuhn** [comment]
:+1: thanks
